common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java

```diff
@@ -218,21 +218,18 @@ public ShuffleMetrics() {
     private final int shuffleId;
     // An array containing mapId and reduceId pairs.
     private final int[] mapIdAndReduceIds;
+    private final int stageAttemptId;
```

Comment:
some more thoughts. Although we use `stageAttempId` as the `shuffleGenerationId`, but they are 2 different concepts. it seems better to name it `shuffleGenerationId`.

Reply:
Thanks, shuffleGenerationId is a more precise name, I'll change all related place.